File: 7-kyu/closest-to-zero/javascript/solution.js
const closest = (arr) => {
  let smallestPos = Number.MAX_SAFE_INTEGER;
  let smallestNeg = Number.MAX_SAFE_INTEGER;
  let zeroFound = false;
  for (const num of arr) {
    if (num > 0) {
      smallestPos = Math.min(smallestPos, num);
    } else if (num < 0) {
      smallestNeg = Math.abs(num) < Math.abs(smallestNeg) ? num : smallestNeg;
    } else {
      zeroFound = true;
    }
  }
  let closestDiff;
  if (zeroFound) {
    closestDiff = 0;
  } else if (Math.abs(smallestPos) < Math.abs(smallestNeg)) {
    closestDiff = smallestPos;
  } else if (Math.abs(smallestPos) > Math.abs(smallestNeg)) {
    closestDiff = smallestNeg;
  } else {
    closestDiff = null;
  }
  return closestDiff;
}

File: 7-kyu/closest-to-zero/python/solution.py
def closest(lst):
    if 0 in lst:
        return 0
    smallest_diff = None
    closest_to = lst[0]
    for num in lst:
        curr_diff = abs(0 - num)
        if smallest_diff is None:
            smallest_diff = curr_diff
            closest_to = num
        elif curr_diff < smallest_diff:
            smallest_diff = curr_diff
            closest_to = num
    return closest_to if lst.count(-closest_to) == 0 else None
